internal/nodes/dequeue_ingestion.go

```diff
@@ -89,7 +89,7 @@ func DequeueIngestion(key string, fullsync bool) {
 		if found {
 			objects.SharedlbLister().Delete(namespace + "/" + name)
 			utils.AviLog.Infof("key: %s, msg: service transitioned from type loadbalancer to ClusterIP or NodePort, will delete model", name)
-			model_name := lib.GetModelName(lib.GetTenant(), lib.GetNamePrefix()+namespace+"-"+name)
+			model_name := lib.GetModelName(lib.GetTenant(), lib.Encode(lib.GetNamePrefix()+namespace+"-"+name, lib.Service))
```

Comment:
Not sure about this. If the service is found in the LB lister then the model that is marked nil will be the L4 service model (as the comment also mentions). So I guess the objectType should be `lib.L4VS` instead of `lib.Service`. This is a l4 VS deletion case.
cc: @monotosh-avi ptal.

Reply:
Yes looks like it should be lib.L4VS